python-frontend/src/test/java/org/sonar/python/tree/PythonTreeMakerTest.java

```diff
@@ -976,6 +1026,49 @@ void classdef_statement() {
     assertThat(classDef.children()).hasSize(7);
   }
 
+  @Test
+  void classdef_statement_type_params() {
+    setRootRule(PythonGrammar.CLASSDEF);
+    var astNode = p.parse("class generic_class[\n" +
+      "   SimpleTypeVar,\n" +
+      "   TypeVarWithBound: int,\n" +
+      "   TypeVarWithConstraints: (str, bytes),\n" +
+      "   *SimpleTypeVarTuple,\n" +
+      "   **SimpleParamSpec\n" +
+      "]: pass");
+    var classDef = treeMaker.classDefStatement(astNode);
```

Comment:
Same here, we can use:
```suggestion
var classDef = parse("class generic_class[\n" +
      "   SimpleTypeVar,\n" +
      "   TypeVarWithBound: int,\n" +
      "   TypeVarWithConstraints: (str, bytes),\n" +
      "   *SimpleTypeVarTuple,\n" +
      "   **SimpleParamSpec\n" +
      "]: pass", treeMaker::classDefStatement);
```
When doing so, we can see that the `TypeParams` are not registered as children of `ClassDef`.

Reply:
Good catch, thanks